src/physics.F90

```diff
@@ -459,11 +470,17 @@ subroutine elastic_scatter(i_nuclide, rxn, E, uvw, mu_lab, wgt)
     vel = sqrt(E)
 
     ! compute cosine of scattering angle in LAB frame by taking dot product of
-    ! neutron's pre- and post-collision angle
-    mu_lab = dot_product(uvw, v_n) / vel
+    ! neutron's pre- and post-collision unit vectors
+    if (iso_lab) then
+      uvw(1) = TWO * prn() - ONE
+      phi = TWO * PI * prn()
+      uvw(2) = cos(phi) * sqrt(ONE - uvw(1)*uvw(1))
+      uvw(3) = sin(phi) * sqrt(ONE - uvw(1)*uvw(1))
```

Comment:
I went ahead and consolidated the iso-in-lab scattering logic into `subroutine scatter` such that the `elastic_scatter`, `sab_scatter` and `inelastic_scatter` routines now remain the same as before. I would certainly like both @paulromano and @walshjon at the least to look at this to make sure that the new (greatly simplified) implementation will still perform as expected.